### PR TITLE
fix: correct exemplar formating

### PR DESCRIPTION
**Pull request by voltbit**
Fix exemplar formating

- Fixed improper formating of exemplars
 - Added snapshot test for exemplars

Fixes https://github.com/siimon/prom-client/issues/554

### CODE REVIEW

File: lib/registry.js [hunk 1]
```diff
@@ -57,19 +58,18 @@ class Registry {
 				? `{${formattedLabels.join(',')}}`
 				: '';
 
-			values.push(
-				`${metricName}${labelsString} ${getValueAsString(val.value)}`,
-			);
+			fullMetricLine = `${metricName}${labelsString} ${getValueAsString(
+				val.value,
+			)}`;
```
Comment (ngavalas) [inline lib/registry.js:63]: nit: why did you declare it above? Should be fine to just `let fullMetricLine = ...` here.

**Review by SimenB — APPROVED** [review]
wonderful, thanks!

**Comment by SimenB** [comment]
I cherry-picked this onto master via 4a800f4e0ec43fc95e33273d0f63499d9347c08c. I think I applied the diff correctly 😅

**Comment by voltbit** [comment]
It looks good, thanks!

**Pull request merged**